internal/controller/openshiftbuild_controller.go

```diff
@@ -152,13 +171,58 @@ func (r *OpenShiftBuildReconciler) CreateOrUpdate(ctx context.Context, client cl
 	})
 }
 
+// ReconcileSharedResource creates and updates SharedResource objects
+func (r *OpenShiftBuildReconciler) ReconcileSharedResource(ctx context.Context, openshiftBuild *openshiftv1alpha1.OpenShiftBuild) error {
+	logger := log.FromContext(ctx).WithValues("name", openshiftBuild.ObjectMeta.Name)
+	SharedResourceState := openshiftBuild.Spec.SharedResource.State
+
+	if SharedResourceState != openshiftv1alpha1.Enabled && SharedResourceState != openshiftv1alpha1.Disabled {
+		return errors.New("unknown component state")
+	}
```

Comment:
Not blocking:

I think it's totally acceptable to set the value to `Enabled` here. I don't think k8s will allow this due to CRD validations, but it doesn't hurt to ensure a proper value is set here.